system/jlib/jdebug.hpp

```diff
@@ -225,6 +225,43 @@ class jlib_decl HiresTimer
     unsigned start;
 };
 
+//---------------------------------------------------------------------------------------------------------------------
+
+/*
+  There are many situations where you want to report or check something periodically.  This class encapsulates th code
+  to perform the check.
+  The constructor has an option to indicate whether the first hasElapsed call should return true, or be suppressed.
+*/
+class PeriodicTimer
+{
+public:
+    PeriodicTimer(unsigned ms, bool suppressFirst)
+    {
+        timePeriodCycles = nanosec_to_cycle(ms * 1000000);
+        lastElapsedCycles = get_cycles_now();
+        if (!suppressFirst)
+            lastElapsedCycles -= timePeriodCycles;
+    }
+
+    bool hasElapsed()
+    {
+        return hasElapsed(get_cycles_now());
+    }
+
+    bool hasElapsed(cycle_t nowCycles)
+    {
+        if ((nowCycles - lastElapsedCycles) < timePeriodCycles)
```

Comment:
minor: as hasElapsed (at least in context I'm thinking of) will be called a relatively high amount of times, 
perhaps better to do the math at start/on elapsed

e.g. change ctor to do:
```
nextElapsedCycles = get_cycles_now();
if (suppressFirst)
    nextElapsedCycles += timePeriodCycles;
```
and change this to:
```
if (nowCycycles < nextElapsedCycles)
    return false;

nextElapsedCycles = nowCycles + timePeriodCycles;
```

Reply:
That has potential problems with overflow - granted it will only happen after several years unless processor speeds increase.